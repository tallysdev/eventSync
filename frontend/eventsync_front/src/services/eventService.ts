import api from '@/services/api'
import { useAuthStore } from '@/stores/auth'
import { AxiosError } from 'axios'

export const fetchEvents = (
  page: number = 1,
  pageSize: number = 10,
  status?: string,
  type?: string,
  name?: string
) => {
  return api.get('events/', {
    params: { page, page_size: pageSize, status, event_type: type, name }
  })
}

export const fetchEvent = (id: number) => {
  return api.get(`events/${id}/`)
}

export const addEvent = (formData: FormData) => {
  const { token } = useAuthStore()
  return api.post('events/', formData, {
    headers: { Authorization: `Bearer ${token}` }
  })
}

export const fetchLocations = () => {
  return api.get('locals')
}

export const fetchLocal = (id: number) => {
  return api.get(`locals/${id}/`)
}

export const signupForEvent = async (eventId: number, userId: number) => {
  const { token } = useAuthStore()
  try {
    const response = await api.post(`events/registration/`, null, {
      params: { event_id: eventId, user_id: userId },
      headers: { Authorization: `Bearer ${token}` }
    })
    return response
  } catch (error) {
    const axiosError = error as AxiosError

    console.error('Error signing up for event:', axiosError.response?.data || axiosError.message)
    throw error
  }
}

export const cancelSubscription = async (eventId: number, userId: number) => {
  const { token } = useAuthStore()
  try {
    const response = await api.delete(`events/registration/${eventId}/${userId}/`, {
      headers: { Authorization: `Bearer ${token}` }
    })
    return response
  } catch (error) {
    const axiosError = error as AxiosError

    console.error(
      'Error canceling subscription for event:',
      axiosError.response?.data || axiosError.message
    )
    throw error
  }
}

export const checkUserSubscription = async (eventId: number, userId: number) => {
  const { token } = useAuthStore()
  try {
    const response = await api.get(`events/registration/${eventId}/${userId}/`, {
      headers: { Authorization: `Bearer ${token}` }
    })
    return response
  } catch (error) {
    const axiosError = error as AxiosError

    console.error(
      'Error checking subscription status:',
      axiosError.response?.data || axiosError.message
    )
    throw error
  }
}

export const fetchMyEvents = (
  page: number = 1,
  pageSize: number = 10,
  status?: string,
  type?: string,
  name?: string
) => {
  const { token } = useAuthStore()
  return api.get('eventsorganized/', {
    params: { page, page_size: pageSize, status, event_type: type, name },
    headers: { Authorization: `Bearer ${token}` } 
  })
}

export const fetchEventsPresence = (
  page: number = 1,
  pageSize: number = 10,
  status?: string,
  type?: string,
  name?: string
) => {
  const { token } = useAuthStore()
  return api.get('eventspresence/', {
    params: { page, page_size: pageSize, status, event_type: type, name },
    headers: { Authorization: `Bearer ${token}` } 
  })
}

export const updateEvent = (formData: FormData, eventId: number) => {
  const { token } = useAuthStore()
  return api.patch(`events/${eventId}/`, formData, {
    headers: { Authorization: `Bearer ${token}` }
  })
}

export const deleteEvent = async (eventId: number) => {
  const { token } = useAuthStore()
  try {
    const response = await api.delete(`events/${eventId}/`, {
      headers: { Authorization: `Bearer ${token}` }
    })
    return response
  } catch (error) {
    const axiosError = error as AxiosError

    console.error(
      'Error deleting event:',
      axiosError.response?.data || axiosError.message
    )
    throw error
  }
}

export const getOrganizer = (eventid: number) => {
  const { token } = useAuthStore()
  return api.get(`organizer/${eventid}/`, {
    headers: { Authorization: `Bearer ${token}` }
  })
}