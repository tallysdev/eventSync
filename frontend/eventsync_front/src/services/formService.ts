import api from '@/services/api'

export const addForm = (formData: FormData) => {
  return api.post('forms/', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}
