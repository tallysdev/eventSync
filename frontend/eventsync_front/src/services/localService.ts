import api from '@/services/api'
import { type Local } from '@/types/local'
import { useAuthStore } from '@/stores/auth'

export const fetchLocations = (page: number, pageSize: number) => {
  return api.get('locals/', {
    params: { page, page_size: pageSize }
  })
}

export const addLocal = (formData: FormData) => {
  const { token } = useAuthStore()
  return api.post('locals/', formData, {
    headers: { Authorization: `Bearer ${token}` }
  })
}

export const fetchLocal = () => {
  return api.get('locals')
}

export const updateLocal = async (local: Local) => {
  const { token } = useAuthStore()
  return api.put(`locals/${local.id}/`, local, {
    headers: { Authorization: `Bearer ${token}` }
  })
}

export const deleteLocal = async (id: number) => {
  const { token } = useAuthStore()
  return api.delete(`locals/${id}/`, {
    headers: { Authorization: `Bearer ${token}` }
  })
}
