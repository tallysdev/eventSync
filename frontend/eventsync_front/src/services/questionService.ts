import api from '@/services/api'

export const addQuestions = (formData: FormData) => {
  return api.post('questions/', formData, {
    headers: { 'Content-Type': 'multipart/question-data' }
  })
}
