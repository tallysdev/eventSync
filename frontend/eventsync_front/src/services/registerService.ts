import api from '@/services/api'
import { type RegisterUser } from '@/types/users'

export const registerUserInDB = (user: RegisterUser) => {
  return api.post('register/', user)
}
