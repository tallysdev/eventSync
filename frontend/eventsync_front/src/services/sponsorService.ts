// services/sponsorService.ts
import api from '@/services/api'

export const fetchSponsors = (page: number, pageSize: number) => {
  return api.get('sponsors/', {
    params: { page, page_size: pageSize }
  })
}

export const addSponsor = (formData: FormData) => {
  // TODO: add authentication
  return api.post('sponsors/', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}
