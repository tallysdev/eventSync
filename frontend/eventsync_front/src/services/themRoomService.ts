import api from '@/services/api'
import { useAuthStore } from '@/stores/auth'

export const fetchThemeRooms = (eventId: number) => {
    return api.get('themeRoom/', {
        params: { event_id: eventId }
    })
}

export const fetchThemeRoom = (id: number) => {
    return api.get(`themeRoom/${id}/`)
}

export const addThemeRoom = (formData: FormData) => {
    const { token } = useAuthStore()
    return api.post('themeRoom/', formData, {
        headers: { Authorization: `Bearer ${token}` }
    })
}

export const updateThemeRoom = (id: number, formData: FormData) => {
    const { token } = useAuthStore()
    return api.patch(`themeRoom/${id}/`, formData, {
        headers: { Authorization: `Bearer ${token}` }
    })
}

export const deleteThemeRoom = (id: number) => {
    const { token } = useAuthStore()
    return api.delete(`themeRoom/${id}/`, {
        headers: { Authorization: `Bearer ${token}` }
    })
}
