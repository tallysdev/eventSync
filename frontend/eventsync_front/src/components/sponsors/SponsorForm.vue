<template>
  <v-dialog :model-value="dialog" @update:model-value="updateDialog" max-width="600">
    <v-card>
      <v-card-title>
        <span class="headline">Novo Patrocinador</span>
      </v-card-title>
      <v-card-text>
        <v-form ref="form" v-model="valid" lazy-validation>
          <v-text-field
            v-model="newSponsor.name"
            label="Nome"
            :rules="[rules.required]"
            required
          ></v-text-field>
          <v-text-field
            v-model="newSponsor.email"
            label="Email"
            :rules="[rules.required, rules.email]"
            required
          ></v-text-field>
          <v-text-field
            v-model="newSponsor.phone"
            label="Telefone"
            :rules="[rules.required]"
            required
          ></v-text-field>
          <v-textarea
            v-model="newSponsor.description"
            label="Descrição"
            :rules="[rules.required]"
            required
          ></v-textarea>
          <v-file-input
            v-model="newSponsor.logo"
            label="Logo"
            :rules="[rules.required]"
            required
          ></v-file-input>
        </v-form>
        <v-alert v-if="errorMessage" type="error" dismissible v-model="showError">
          {{ errorMessage }}
        </v-alert>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click="closeDialog">Cancelar</v-btn>
        <v-btn color="blue darken-1" text @click="addSponsorData" :disabled="!valid">Salvar</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script setup lang="ts">
import { addSponsor } from '@/services/sponsorService'
import { emailValidation, requiredValidation } from '@/utils/validation'
import { ref, watch } from 'vue'
import { type Sponsor } from '@/types/sponsor'

const props = defineProps({
  dialog: {
    type: Boolean,
    required: true
  }
})

const emit = defineEmits(['update:dialog', 'sponsor-added'])

const rules = {
  required: requiredValidation,
  email: emailValidation
}

const valid = ref(false)
const newSponsor = ref<Sponsor>({
  name: '',
  email: '',
  phone: '',
  description: '',
  logo: ''
})

const errorMessage = ref<string | null>(null)
const showError = ref(false)

const closeDialog = () => {
  emit('update:dialog', false)
}

const updateDialog = (value: boolean) => {
  emit('update:dialog', value)
}

const addSponsorData = async () => {
  const formData = new FormData()
  formData.append('name', newSponsor.value.name)
  formData.append('email', newSponsor.value.email)
  formData.append('phone', newSponsor.value.phone)
  formData.append('description', newSponsor.value.description)
  formData.append('logo', newSponsor.value.logo || '')

  try {
    await addSponsor(formData)
    closeDialog()
    newSponsor.value = {
      name: '',
      email: '',
      phone: '',
      description: '',
      logo: ''
    }
    emit('sponsor-added', 'Patrocinador adicionado com sucesso!')
  } catch (error) {
    console.error(error)
    errorMessage.value =
      'Ocorreu um erro ao tentar adicionar o patrocinador. Por favor, tente novamente.'
    showError.value = true
  }
}

watch(
  () => props.dialog,
  (newValue) => {
    if (!newValue) {
      newSponsor.value = {
        name: '',
        email: '',
        phone: '',
        description: '',
        logo: ''
      }
      showError.value = false
      errorMessage.value = null
    }
  }
)
</script>

<style scoped></style>
