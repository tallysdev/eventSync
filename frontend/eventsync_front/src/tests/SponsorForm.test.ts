import { mount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import SponsorForm from '@/components/sponsors/SponsorForm.vue'
import { addSponsor } from '@/services/sponsorService'

vi.mock('@/services/sponsorService', () => ({
  addSponsor: vi.fn()
}))

describe('SponsorForm.vue', () => {

  it('render correctly', () => {
    const wrapper = mount(SponsorForm, {
      props: {
        dialog: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the form correctly', () => {
    const wrapper = mount(SponsorForm, {
      props: {
        dialog: true
      }
    })

    expect(wrapper.find('v-text-field[label="Nome"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Email"]').exists()).toBe(true)
    expect(wrapper.find('v-text-field[label="Telefone"]').exists()).toBe(true)
    expect(wrapper.find('v-textarea[label="Descrição"]').exists()).toBe(true)
    expect(wrapper.find('v-file-input[label="Logo"]').exists()).toBe(true)
  })

  // it('shows an error message when there is an error', async () => {
  //   const errorMessage =
  //     'Ocorreu um erro ao tentar adicionar o patrocinador. Por favor, tente novamente.'
  //   vi.mocked(addSponsor).mockRejectedValue(new Error(errorMessage))

  //   const wrapper = mount(SponsorForm, {
  //     props: {
  //       dialog: true
  //     }
  //   })

  //   await wrapper.setData({
  //     newSponsor: { name: 'Test', email: 'test@example.com', phone: '', description: '', logo: '' }
  //   })
  //   await wrapper.find('v-btn[text="Salvar"]').trigger('click')

  //   expect(wrapper.find('v-alert').text()).toContain(errorMessage)
  // })

  // it('calls addSponsor when form is submitted successfully', async () => {
  //   vi.mocked(addSponsor).mockResolvedValue({
  //     data: undefined,
  //     status: 0,
  //     statusText: '',
  //     headers: undefined,
  //     config: undefined
  //   })

  //   const wrapper = mount(SponsorForm, {
  //     props: {
  //       dialog: true
  //     }
  //   })

  //   await wrapper.setData({
  //     newSponsor: { name: 'Test', email: 'test@example.com', phone: '', description: '', logo: '' }
  //   })
  //   await wrapper.find('v-btn[text="Salvar"]').trigger('click')

  //   expect(vi.mocked(addSponsor).toHaveBeenCalled())
  // })

  // it('emits update:dialog when the dialog is closed', async () => {
  //   const wrapper = mount(SponsorForm, {
  //     props: {
  //       dialog: true
  //     }
  //   })

  //   await wrapper.find('v-btn[text="Cancelar"]').trigger('click')

  //   expect(wrapper.emitted('update:dialog')).toBeTruthy()
  //   expect(wrapper.emitted('update:dialog')?.[0]).toEqual([false])
  // })
})
