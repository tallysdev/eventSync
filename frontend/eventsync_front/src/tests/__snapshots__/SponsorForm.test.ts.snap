// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SponsorForm.vue > render correctly 1`] = `
"<v-dialog model-value="true" max-width="600">
  <v-card>
    <v-card-title><span class="headline">Novo Patrocinador</span></v-card-title>
    <v-card-text>
      <v-form modelvalue="false" lazy-validation="">
        <v-text-field modelvalue="" label="Nome" rules="(value) => {
  if (!value) {
    return &quot;Campo obrigatório&quot;;
  }
  return true;
}" required=""></v-text-field>
        <v-text-field modelvalue="" label="Email" rules="(value) => {
  if (!value) {
    return &quot;Campo obrigatório&quot;;
  }
  return true;
},(value) => {
  if (!value) {
    return &quot;Email é necessário&quot;;
  }
  const emailRegex = /^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$/;
  if (!emailRegex.test(value)) {
    return &quot;Por favor, insira um email válido&quot;;
  }
  return true;
}" required=""></v-text-field>
        <v-text-field modelvalue="" label="Telefone" rules="(value) => {
  if (!value) {
    return &quot;Campo obrigatório&quot;;
  }
  return true;
}" required=""></v-text-field>
        <v-textarea modelvalue="" label="Descrição" rules="(value) => {
  if (!value) {
    return &quot;Campo obrigatório&quot;;
  }
  return true;
}" required=""></v-textarea>
        <v-file-input modelvalue="" label="Logo" rules="(value) => {
  if (!value) {
    return &quot;Campo obrigatório&quot;;
  }
  return true;
}" required=""></v-file-input>
      </v-form>
      <!--v-if-->
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn color="blue darken-1" text="">Cancelar</v-btn>
      <v-btn color="blue darken-1" text="" disabled="true">Salvar</v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>"
`;
